File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Mauritius.m
//
//  UIImage+Mauritius.m
//  CountriesFlags
//

#import "Countries/UIImage+Mauritius.h"

@implementation UIImage (Mauritius)

+ (UIImage * _Nullable)flagMauritiusWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:235.0f/255.0f
                                        green:36.0f/255.0f
                                         blue:54.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorBlue = [UIColor colorWithRed:19.0f/255.0f
                                         green:26.0f/255.0f
                                          blue:109.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:1.0f
                                           green:214.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:166.0f/255.0f
                                           blue:80.0f/255.0f
                                          alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 4.0f, size.width, size.height / 4.0f * 3.0f));
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 4.0f * 2.0f, size.width, size.height / 4.0f * 2.0f));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 4.0f * 3.0f, size.width, size.height / 4.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Mauritius.swift
//
//  UIImage+Mauritius.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Mauritius "wikipedia"
     
     Generates the flag of Mauritius.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Mauritius.
     */
    internal static func flagMauritius(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 235.0 / 255.0,
                               green: 36.0 / 255.0,
                               blue: 54.0 / 255.0,
                               alpha: 1.0)
        let colorBlue = UIColor(red: 19.0 / 255.0,
                                green: 26.0 / 255.0,
                                blue: 109.0 / 255.0,
                                alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 214.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)
        let colorGreen = UIColor(red: 0.0,
                                 green: 166.0 / 255.0,
                                 blue: 80.0 / 255.0,
                                 alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 4.0),
                          size: CGSize(width: size.width, height: size.height / 4.0 * 3.0)))

        colorYellow.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 4.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 4.0 * 2.0)))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 4.0 * 3.0),
                          size: CGSize(width: size.width, height: size.height / 4.0)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
